src/offset/mod.rs

```diff
@@ -556,4 +574,23 @@ mod tests {
         Utc.timestamp_nanos(i64::default());
         Utc.timestamp_nanos(i64::min_value());
     }
+
+    #[test]
+    fn test_negative_micros() {
+        let dt = Utc.timestamp_micros_opt(-1_000_000).unwrap();
+        assert_eq!(dt.to_string(), "1969-12-31 23:59:59 UTC");
+        let dt = Utc.timestamp_micros_opt(-999_999).unwrap();
+        assert_eq!(dt.to_string(), "1969-12-31 23:59:59.000001 UTC");
+        let dt = Utc.timestamp_micros_opt(-1).unwrap();
+        assert_eq!(dt.to_string(), "1969-12-31 23:59:59.999999 UTC");
+        let dt = Utc.timestamp_micros_opt(-60_000_000).unwrap();
+        assert_eq!(dt.to_string(), "1969-12-31 23:59:00 UTC");
+        let dt = Utc.timestamp_micros_opt(-3_600_000_000).unwrap();
+        assert_eq!(dt.to_string(), "1969-12-31 23:00:00 UTC");
+    }
+
+    #[test]
+    fn test_micros_example() {
```

Comment:
I don't think you need to copy the doctest here.

Reply:
Thanks, I wasn't aware this gets tested automatically :)